api/routers/groups.py

```diff
@@ -0,0 +1,60 @@
+from api.models.devices import Device
+from api.models.groups import Group
+from asyncpg import UniqueViolationError
+from fastapi import APIRouter, HTTPException, status
+
+router = APIRouter(
+    prefix="/groups",
+    tags=["groups"],
+)
+
+
+@router.post("", response_model=Group)
+async def add_group(group: Group):
+    """Add a group."""
+    try:
+        return await Group.add(group)
+    except UniqueViolationError:
+        raise HTTPException(status_code=status.HTTP_409_CONFLICT, detail="Group name already exists.")
+
+
+@router.get("", response_model=list[Group])
+async def get_groups():
+    """Get a list of all groups."""
+    return await Group.get_all()
+
+
+@router.get("/{id}", response_model=Group)
+async def get_group_id(id: int):
+    """Get a group by id."""
+    group = await Group.get(id)
+    if not group:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="No group with that id was found.")
```

Comment:
Avec mon commentaire plus haut on pourrait avoir:

```py
try:
    group = await Group.get(id)
except ValueError as exc:
    raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="No group with that id was found.") from exc
```

IMHO perso je trouve ça plus secure que de vérifier `if group is not None` à chaque fois.

Reply:
On fonctionne avec None dans le reste de l'api, mais c'est assez intéressant de gérer ça avec des exceptions.